frontend/src/api/TokenRefreshInstance.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { FIVETHOUSAND, THOUSAND } from '../constants/valuableNumbers'
+import {HttpClient} from './HttpClient'
+
+interface IToken {
+    access:string
+token: {
+    access: string
+    raw: string
+    payload: {
+        exp: number
+        id: number
+    }
+}
+split: any
+}
+
+interface ITokenSend {
+    refresh: string | null
+}
+
+interface ITokenRefresh {
+    access: string
+    refresh: string
+}
+
+export interface ISendTokenRefresh {
+    postRefresh: (payload: ITokenSend) => Promise<ITokenRefresh>
+}
+
+export function jwtDecode(t: IToken) {
+    const token = {
+        raw: {},
+        payload: {},
+    }
+    token.raw = t
+    token.payload = JSON.parse(window.atob(t.split('.')[1]))
```

Comment:
What will happened if some data is missing in parsing result? For instance, if it does not contain id or expiration time?